Add ExtThrust for more explicit dependencies

sethrj opened the pull request:
This is a proposed solution to the problem motivating #1692  (see https://github.com/celeritas-project/celeritas/pull/1692#issuecomment-2745408133).  ~~@mrguilima @whokion @esseivaju could one of you please see if this works?~~

I think this is safe to merge; it simply makes the dependencies clearer.

sethrj committed:
Add ExtThrust and include with tests/libs that use it

sethrj committed:
Remove hopefully now-unnecessary ldl

esseivaju commented:
I haven't needed the explicit `ldl` in my CMakePreset, since https://github.com/celeritas-project/celeritas/pull/1661

mrguilima commented:
I also haven't seen the problem for a while.  I thought the problem was fixed by some other commit, but then I was surprised that @whokion was seeing it.  I hope he can test it.

sethrj committed:
Remove thrust link

```
CMake Error: install(EXPORT "celeritas-targets" ...) includes target "ExtThrust" which requires target "_Thrust_Thrust" that is not in any export set.
```

weird

sethrj committed:
Move roctx_library search to top-level cmakelists

sethrj committed:
Merge remote-tracking branch 'upstream/develop' into ext-thrust

pcanal commented:
Are those libraries already depending on `ExtThurst`?   Does the build still suceed if `Thrust` is not installed?

sethrj commented:
@pcanal these all depend on thrust if cuts or rocm is enabled, but most distributions include them in the paths by default so don't need to explicitly link.

pcanal reviewed (approved):
I agree this can be merged even-though it still feels 'weird' to have to be this explicit.

sethrj commented:
I think @wdconinc saw on some spack setups that an explicit thrust dependency was needed?
